### DISCUSSION TITLE
git error: fatal: Invalid path '/storage/emulated': No such file or directory

### DISCUSSION THREAD

**Discussion by sudoexec** [termux/termux-app#3537] [Q&A] [answered]
I want to clone a repo to /data/data/com.termu/files/home/storage/shared/repo, but git show error:
`fatal: Invalid path '/storage/emulated': No such file or directory`

I've setup storage permissions with `termux-setup-storage`, and I have access to `/data/data/com.termu/files/home/storage/shared` directory.

Android Version: Android 13
Termux Version: 0.118.0 from F-Droid

**Comment by sylirre** [comment, marked as answer]
What is the command you using for git clone? It looks like you are doing something wrong.

For example `git clone https://github.com/termux/termux-app ~/storage/shared/termux-app` works without issues:

![Screenshot_20230804-120152](https://github.com/termux/termux-app/assets/107305601/a138d589-7389-4f8f-b966-24f92b31692d)

Anyway, I would not recommend working with git repositories on shared storage. Just the file system on shared storage is quite restricted in features (it's an emulated layer for FAT32/exFAT compatibility) and `git` tends to have problems (object corruption, etc) when working with repositories here.

File managers and code editors are capable of accessing files on storage volumes and that means you can access your repository when it is placed in Termux home directory: https://wiki.termux.com/wiki/Internal_and_external_storage#Access_Termux_from_a_file_manager

**Reply by sudoexec** [reply]
Thanks for your help.I tried to reinstall many times, and somehow it works now.
In fact, take security into account, I don't want to put my git repo to shared storage. But some apps don't support opening external storage(like Obsidian).
